### PR TITLE
Add version macros for 0.2 release and prefer g_clear_object to g_object_unref.

### PR DESCRIPTION
**Pull request by Kurt-von-Laven**
We forgot to add the new version macros for the 0.2 release. Better late
than never.

g_clear_object is safer because it NULLs out the reference to the object
being unreffed, and the performance penalty is miniscule.

[endlessm/eos-sdk#2906]

### CODE REVIEW

**Comment by ptomato** [comment]
Interesting conclusion. It's sad to lose the singleton from a code convenience standpoint. Also it's kind of unfortunate that the metrics might not be sent over the bus until you `g_object_unref` the recorder object. And this is actually quite a problem for recording metrics within Javascript code; because the recorder might not get released until a much later garbage collection pass.

Because of this I'd like to be absolutely sure that this is the best solution. Is flushing necessary in general, or is it really only necessary in shutdown where it's a matter of getting the jump on those extra microseconds? Could we add an `emtr_event_recorder_flush()` method (possibly even undocumented?) and use it only in eos-metrics-instrumentation? Could we get away with flushing after 
every event recorder call?

On a side note: strictly speaking we should bump the version right now to the 0.3 development series, but since we will go to 0.4 very shortly I think we can skip that.

**Comment by ptomato** [comment]
Or, maybe we could get away with not deprecating `emtr_event_recorder_get_default()` but instead add a note saying that if you need control over when it is flushed, use `emtr_event_recorder_new()` instead. Although that is basically equivalent to adding a flush method but without actually adding new API.

**Comment by cosimoc** [comment]
FWIW I agree with @ptomato this sounds a bit overkill. Why can't `eos-metrics-instrumentation` just flush the dbus connection before quitting?
- it's weird for library code to implicitly flush a dbus connection
- event recorder suddenly became quite expensive, since destroying and recreating one will flush dbus synchronously. I am a little concerned of having this in e.g. mutter

File: eosmetrics/emtr-event-recorder.c [hunk 1]
```diff
@@ -117,14 +117,31 @@ emtr_event_recorder_finalize (GObject *object)
...
+        g_dbus_connection_flush_sync (dbus_connection, NULL /* GCancellable */,
+                                      &error);
+      if (!flush_succeeded)
+        {
+          g_warning ("Failed to flush DBus connection to client-side daemon: "
+                     "%s.", error->message);
+          g_error_free (error);
+        }
+
+      g_object_unref (priv->dbus_proxy);
```
Comment (ptomato) [inline eosmetrics/emtr-event-recorder.c]: Use `g_clear_object`

**Comment by Kurt-von-Laven** [comment]
`*_get_default` works in practice for almost all situations, but I'm pretty averse to giving consumers of the event recorder API options (e.g., Do I use `*_get_default` or `*_new`? Should I call `*_flush`? If so, how often?) when one way can work for every use case we have seen so far. The consistency increases the chances that when someone adds a new event recorder call by aping an existing one, it will do what they want without more thinking required. The "power" API is the DBus API exposed by the event recorder daemon itself; the event recorder's job is to make using that daemon as mindless as possible. I'd rather only add a `*_flush` method once we have a concrete need for one (e.g., recording a shutdown event from GJS land) because that will keep the event recorder API easier to use in the meantime.

With regards to flushing each event recording individually Roddy, was very concerned with any event recording calls (excluding initialization and finalization) being synchronous because he doesn't want the user experience to be affected by our data gathering activities. If we flushed asynchronously instead, then we would still have the same problem with shutdown events potentially being dropped in eos-metrics-instrumentation.

Flushing in a finalize method doesn't seem weird to me. Plenty of libraries do this. For example, streams are often flushed when released because it spares consumers of those APIs from thinking about flushing if they don't have to.

Creating an event recorder has always been expensive because it synchronously connects to the event recorder daemon. That's why we don't do it very often (e.g., https://github.com/endlessm/mutter/pull/29/files). We could eventually optimize this, but I don't think it's worth the extra complexity of making the code asynchronous at this point given how seldom it occurs.

**Comment by Kurt-von-Laven** [comment]
Turns out there is a way to flush the system DBus connection from eos-metrics-instrumentation without modifying the event recorder's API. Woohoo!

File: eosmetrics/emtr-event-recorder.c [hunk 2]
```diff
@@ -123,7 +123,7 @@ emtr_event_recorder_finalize (GObject *object)
   g_variant_unref (priv->empty_auxiliary_payload);
 
   if (priv->dbus_proxy != NULL)
-    g_object_unref (priv->dbus_proxy);
+    g_clear_object (&priv->dbus_proxy);
```
Comment (cosimoc) [inline eosmetrics/emtr-event-recorder.c]: No need for the `NULL` check if you're using `g_clear_object()`

**Comment by cosimoc** [comment]
@Kurt-von-Laven this is Revise ready

**Comment by Kurt-von-Laven** [comment]
@cosimoc Ty; CR ready.

**Pull request merged**